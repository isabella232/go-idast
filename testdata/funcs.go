package vars

func A(b, c string, d int) (w, x float, y bool) {
	return len(b+c) + d
}
