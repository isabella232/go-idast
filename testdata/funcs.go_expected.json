 File            | package vars\n\nfunc A(b, c str | vars.go
 Ident           | vars                            | vars.go/Name/Ident
 FuncDecl        | func A(b, c string, d int) (w,  | vars.go/Decls/0/FuncDecl
 Ident           | A                               | vars.go/Decls/0/FuncDecl/Name/Ident
 FuncType        | func(b, c string, d int) (w, x  | vars.go/Decls/0/FuncDecl/Type/FuncType
 FieldList       | (n/a)                           | vars.go/Decls/0/FuncDecl/Type/FuncType/Params/FieldList
 Field           | (n/a)                           | vars.go/Decls/0/FuncDecl/Type/FuncType/Params/FieldList/List/0/Field
 Ident           | b                               | vars.go/Decls/0/FuncDecl/Type/FuncType/Params/FieldList/List/0/Field/Names/b/Ident
 Ident           | c                               | vars.go/Decls/0/FuncDecl/Type/FuncType/Params/FieldList/List/0/Field/Names/c/Ident
 Ident           | string                          | vars.go/Decls/0/FuncDecl/Type/FuncType/Params/FieldList/List/0/Field/Type/Ident
 Field           | (n/a)                           | vars.go/Decls/0/FuncDecl/Type/FuncType/Params/FieldList/List/1/Field
 Ident           | d                               | vars.go/Decls/0/FuncDecl/Type/FuncType/Params/FieldList/List/1/Field/Names/d/Ident
 Ident           | int                             | vars.go/Decls/0/FuncDecl/Type/FuncType/Params/FieldList/List/1/Field/Type/Ident
 FieldList       | (n/a)                           | vars.go/Decls/0/FuncDecl/Type/FuncType/Results/FieldList
 Field           | (n/a)                           | vars.go/Decls/0/FuncDecl/Type/FuncType/Results/FieldList/List/0/Field
 Ident           | w                               | vars.go/Decls/0/FuncDecl/Type/FuncType/Results/FieldList/List/0/Field/Names/w/Ident
 Ident           | x                               | vars.go/Decls/0/FuncDecl/Type/FuncType/Results/FieldList/List/0/Field/Names/x/Ident
 Ident           | float                           | vars.go/Decls/0/FuncDecl/Type/FuncType/Results/FieldList/List/0/Field/Type/Ident
 Field           | (n/a)                           | vars.go/Decls/0/FuncDecl/Type/FuncType/Results/FieldList/List/1/Field
 Ident           | y                               | vars.go/Decls/0/FuncDecl/Type/FuncType/Results/FieldList/List/1/Field/Names/y/Ident
 Ident           | bool                            | vars.go/Decls/0/FuncDecl/Type/FuncType/Results/FieldList/List/1/Field/Type/Ident
 BlockStmt       | {\n}                            | vars.go/Decls/0/FuncDecl/Body/BlockStmt
